_source/_posts/2022-01-18-deploy-dotnet-container-aws-fargate.md

```diff
@@ -0,0 +1,640 @@
+
+```bash
+dotnet new page -n Chat
+```
+
+Edit **Pages/Shared/\_Layout.cshtml** and modify the second nav item from this:
+
+```html
+<a class="nav-link text-dark" asp-area="" asp-page="/Privacy">Privacy</a>
+```
+
+To this:
+
+```html
+<a class="nav-link text-dark" asp-area="" asp-page="/Chat">Chat</a>
+```
+
+At this point, if you run the `dotnet run` command or run the application from Visual Studio, you should be able to navigate to the chat page that just has the heading **"Chat"**. Make sure that it runs successfully before moving on to the next step.
+
+### Add Docker Support to Your .NET Chat Application
+
+Before you add the chat functionality, add the Docker support. Add a file named **Dockerfile** to the root **Okta.Blog.Chat** folder with below content in it.
+
+```docker
+# build and publish the app
+FROM mcr.microsoft.com/dotnet/sdk:3.1-bullseye AS build
+WORKDIR /src
+## copy csproj and restore as distinct layers
+COPY Okta.Blog.Chat.csproj ./
+RUN dotnet restore
+## copy everything else and publish app
+COPY . ./
+RUN dotnet publish -c release -o /app --no-restore
+
+# final stage/image
+FROM mcr.microsoft.com/dotnet/aspnet:3.1-bullseye-slim AS base
+WORKDIR /app
+COPY --from=build /app ./
+ENTRYPOINT ["dotnet", "Okta.Blog.Chat.dll"]
+```
+
+What is going on is we are building and publishing our application first. Next, we copy the build files and call `ENTRYPOINT`, which translates to "run dotnet with our published DLL" - nothing extraordinary, right? The `FROM` keywords are using other built docker files as base images to build our image from, so all the work of installing .NET etc., is already done.
+
+Now add a **.dockerignore** file with the below content so that they are not copied to the final image.
+
+```
+**/.dockerignore
+**/.git
+**/.vscode
+**/bin
+**/obj
+```
+
+If you are using Visual Studio, you'll see you now have the ability to debug right into a running container in your debug toolbar. If you press **F5** or click the play button, you'll run your app in a Docker Container.
+
+{% img blog/dotnet-aws-ecs-fargate/11-DebugDocker.PNG alt:"VS Docker debug" width:"800" %}{: .center-image }
+
+If you are not using Visual Studio, run the following command to build and start the container.
+
+```bash
+docker build . -t okta-chat
+docker run -it --rm -p 5000:80 okta-chat
+```
+
+The app should be accessible at `http://localhost:5000/`.
+
+### Authorize Your Chat App
+
+Modify **Setup.cs** and uncomment the following line.
+
+```cs
+options.Conventions.AuthorizePage("/Chat");
+```
+
+> **Note**: At this point, if you try to authenticate the `/Chat` path using Google Chrome, you'll get an error. This is because we are running the app without TLS and Chrome blocks set-cookie headers with `SameSite=None` when the `Secure` attribute is not present, which will be preset only for HTTPS requests.
```

Comment:
```suggestion
> **Note**: At this point, if you try to authenticate the `/Chat` path using Google Chrome, you'll get an error. This is because we are running the app without TLS and Chrome blocks set-cookie headers with `SameSite=None` when the `Secure` attribute is not present (which will be preset only for HTTPS requests).
```
clarity